File: DSA/SakshamKashyap_590014183/day73/day73_Question_1.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def sum_of_nodes(root):
    if not root:
        return 0
    return root.val + sum_of_nodes(root.left) + sum_of_nodes(root.right)

# Example usage:
if __name__ == "__main__":
    # Tree: 1 / \ 2 3
    root1 = TreeNode(1, TreeNode(2), TreeNode(3))
    print(sum_of_nodes(root1))  # Output: 6

    # Tree: 10 / 5
    root2 = TreeNode(10, TreeNode(5))
    print(sum_of_nodes(root2))  # Output: 15

File: DSA/pranavdixit_590014418/Day_73_Question_1.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node* left;
    struct Node* right;
};

struct Node* newNode(int data) {
    struct Node* node = (struct Node*)malloc(sizeof(struct Node));
    node->data = data;
    node->left = node->right = NULL;
    return node;
}

int sumOfNodes(struct Node* root) {
    if (root == NULL) return 0;
    return root->data + sumOfNodes(root->left) + sumOfNodes(root->right);
}

int main() {
    // Example 1
    struct Node* root1 = newNode(1);
    root1->left = newNode(2);
    root1->right = newNode(3);
    printf("Sum of nodes: %d\n", sumOfNodes(root1)); // 6

    // Example 2
    struct Node* root2 = newNode(10);
    root2->left = newNode(5);
    printf("Sum of nodes: %d\n", sumOfNodes(root2)); // 15

    return 0;
}
